### PR TITLE
docs: address onboarding gaps

### PR DESCRIPTION
**Pull request by shiyuann**
## Description

This PR fixes the issues that are brought up by @hoang-determined-ai 👍 

1. Write porting guides for different frameworks and put it in Training: Implement Training APIs. 
2. Move porting guide from tutorials to Training: Implement Training APIs 
- Remove pining GPUs 
- Metric reducers should be advanced sections of porting guides (lightly) and references (detailed) 
- Porting guides should include additional best practices and links out to more information in the appropriate sections: 
- Using custom loss, metrics (custom functions) 
- Using custom callbacks 
- Data: Working with different data sources (can link out to “Prepare data”) 
- Validation
- Using Metrics Reducer (can link out) 
2. Tutorials: Remove MNIST tutorials for Keras and Pytorch 
3. Add Concepts page 
4. Categorize Features in the Features page 
5. Move Notebooks, Shells, Commands, TensorBoards, Templates out of Features as a standalone first-level page. And put them under the Additional Capabilities title section. 
6. Move experiment page to be a section of Training: Implement Training APIs. 
- Tell about experiments. 
- Link to experiment cycle. 
- Why you need to implement training APIs? What is the purpose of it? 
- How to submit an experiment. 
7. Add non-clickable Training title section in table of contents. 
8. Add non-clickable SysAdmin title section in table of contents. 
9. Prepare Data (page): Revised it to be opinionated about option A. Just reordering should be fine. 
10. Interact with Cluster: Tell users that Python APIs and REST APIs are only for automation. 
11. Recommend what to use for checkpoints in Ecosystem Integration page - REST API & Python APIs 
13. Enforce Quick Start Guide to be “Getting Started with Examples”. And move it to the landing page of documents. 
- Install Docker  “Follow these instructions to … set up Docker if you haven’t installed it already” 
- Include a note before the gray example box, after the Docker instructions: 
- “The commands below demonstrate how to start a Determined cluster locally, run an example experiment, and access the Determined Web UI” 
- Install Determined 
- Install Determined locally 
- Note: if you plan to use Jupyter notebooks, set up persistent storage by doing X 
- Advanced: install Determined on-premise, AWS, GCP (links) 
- Comment: “installation instructions” leads to an on-prem only installation page; we should have that be mapped to on-prem first, but also link out to Cloud and Kubernetes 
- Cluster: To install Determined, please follow … 
- CLI: Each user should also install the Determined command-line tools … 
14. Interact with the Cluster 
^ WebUI 
^ CLI 
Simplified version of the CLI page 
export DET_MASTER= 
det s 
link to the CLI page 
15. Training an example model 
^ Single GPU with example model 
^ Distributed training with example model 
^ Hyperparameter search with example model 
Exploring the Determined Web UI 
^ Experiment tracking 
^ Checkpointing 
^ Profiling 
^ Cluster management 
Prepare Data 
16. Add a separate Join the Community page 

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/training-apis/api-pytorch-advanced.txt [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+Determined supports proper reduction of arbitrary training and validation metrics, even during
+distributed training, by allowing users to define custom reducers. Custom reducers can be either a
+function or an implementation of the :class:`determined.pytorch.MetricReducer` interface. See
+:meth:`determined.pytorch.PyTorchTrialContext.wrap_reducer` for more details.
+
+.. _reproducible-dataset:
+
+***********************
+ Reproducible Datasets
```
Comment (rb-determined-ai) [inline docs/training-apis/api-pytorch-advanced.txt:26]: Before, the api reference for pytorch.samplers and this explanation of how and why you would use them were always together.  Them being separated and not linked in any way seems more confusing to me.

Also, now that they are separated, this title makes it sound like having Reproducible Data is an advanced thing, rather than something you only need to concern yourself with when you have a totally custom dataset.

File: docs/training-apis/api-pytorch.txt [hunk 2]
```diff
@@ -1,43 +1,141 @@
...
+               transforms.ToTensor(),
+               self.normalize,
+           ]))
+
+       train_loader = determined.pytorch.DataLoader(
+           train_dataset, batch_size=self.context.get_per_slot_batch_size(), shuffle=True,num_workers=self.context.get_hparam("workers", pin_memory=True))
+       return train_loader
+
+In the function :meth:`~determined.pytorch.PyTorchTrial.train_batch` returns a batch of data in one
+of the following format:
```
Comment (rb-determined-ai) [inline docs/training-apis/api-pytorch.txt:138]: formats should be plural

File: docs/training-apis/api-pytorch-advanced.txt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+.. _pytorch-custom-reducers:
+
+******************
+ Reducing Metrics
+******************
+
+Determined supports proper reduction of arbitrary training and validation metrics, even during
+distributed training, by allowing users to define custom reducers. Custom reducers can be either a
+function or an implementation of the :class:`determined.pytorch.MetricReducer` interface. See
+:meth:`determined.pytorch.PyTorchTrialContext.wrap_reducer` for more details.
```
Comment (rb-determined-ai) [inline docs/training-apis/api-pytorch-advanced.txt:21]: This section was written as an introduction to the rest of the metric reducers API.  Why did you separate the introduction from the contents of the API?  This says basically nothing by itself.

File: docs/training-apis/api-pytorch.txt [hunk 4]
```diff
@@ -1,43 +1,141 @@
...
+#############
+ PyTorch API
+#############
 
-.. _pytorch-trial:
+**********
+ Overview
+**********
+
+This document lists how you should implement the PyTorch API. You need to implement a trial class
```
Comment (rb-determined-ai) [inline docs/training-apis/api-pytorch.txt:9]: This document ~lists how you should implement the PyTorch API~ guides you through training a PyTorch model in Determined, by implementing a PyTorchTrial.

Reason: "listing" is the wrong verb, because when you "list XYZ", the "XYZ" needs to be multiple things.  "how you should implement" is just one thing.

Also, "how you should implement" only makes sense for a plugin, but we also document the Context object, which a user does not have to implement at all.

Also, just after this, you describe the Plugin API itself, but the whole document is not about the plugin API itself, which makes this overview extra confusing.

**Review by rb-determined-ai — COMMENTED** [review]
> @katport This PR removes the MNIST tutorials and merges the porting tutorial, which you wrote, with the Pytorch API page. You might want to read the new PyTorch API document.

What exactly got removed here?  I thought our MNIST tutorials were excellent.

**Review by hoang-determined-ai — APPROVED** [review]

File: docs/training-apis/api-pytorch-advanced.txt (hunk 3 above)
Comment (shiyuann) [reply docs/training-apis/api-pytorch-advanced.txt:21]: This is not separated from the contents of the API. This is to add a brief topic guide for linking reducing metrics to the reference guide of which function or class you should use. In the future, this will add more content about what the concept of metric reducing is and when and why you want to do this. Currently, this is to just be separate from the basic concepts which new users would use to implement the basic functionality of PyTorch API. Send me comments on how you think it should be written and we will add more content in the future.

**Comment by shiyuann** [comment]
Skip the e2e test
